diff --git a/sailfish/subdomain_runner.py b/sailfish/subdomain_runner.py
index d8ce1ae..15cca7b 100644
--- a/sailfish/subdomain_runner.py
+++ b/sailfish/subdomain_runner.py
@@ -1331,6 +1331,8 @@ class SubdomainRunner(object):
 
     def initialize(self):
         self.config.logger.info("Consistent intialization started.")
+        visc = self.config.visc
+        self.config.visc = 1.0/6.0
         try:
             for init_it in xrange(0, self.config.init_iters):
                 self.step(False)
@@ -1347,12 +1349,14 @@ class SubdomainRunner(object):
 
         self.config.logger.info("Initialization phase complete.")
         self._sim.iteration = 0
+        self.config.visc = visc
 
         # TODO: Make it possible to cache all fields and distributions here.
         # Rebuild the compute code for the real run.
         self._initialization = False
+        self._update_compute_code()
         self._prepare_compute_kernels()
-        self._init_compute()
+        self._initial_conditions()
 
     def _handle_geo_updates(self):
         q = self._spec.geo_queue
